pkg/controllers/controller.go

```diff
@@ -54,22 +55,26 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 		provisioner.StatusConditions().MarkTrue(v1alpha1.Active)
 	}
 	// 4. Reconcile
-	if _, err := c.Controller.Reconcile(ctx, resource); err != nil {
+	if result, err := c.Controller.Reconcile(ctx, resource); err != nil {
```

Comment:
We need to return this result in this non error case, too

Reply:
You're totally right. Good catch. If we want to continue draining here, it won't propagate properly. Thanks!